serverless/serverless-azure-functions
# fix: Update release pipeline for automated releases

## What did you implement:

- Made the `release` pipeline automatically triggered on merges to master. There is no full-time team dedicated to this project, so trying to maintain a Git Flow workflow is unreasonable and will slow down features the community needs/wants. For now, the `dev` branch will be retired, and all releases will go through `master`
- Rather than a manual release by clicking the pipeline, the release type will be determined by the prefix of the last commit message as follows:
  - **major** = major release
  - **feat** = minor release
  - **beta** = pre-release
  - anything else = patch release
- Manually generated changelog since pipelines doesn't seem to have access to the full git history
- Removed automatic changelog generation from pipeline



## How did you implement it:

- Check last commit message prefix, set `NPM_RELEASE_TYPE` according to rules stated above
- Removed automated generation of changelog
- Ran changelog generation locally



## How can we verify it:

Open a PR against master with one of the specified prefixes and check the resulting release type



## Todos:

_**Note: Run `npm run test:ci` to run all validation checks on proposed changes**_

       **Validate via `npm run lint`**  
       _Note: Some reported issues can be automatically fixed by running `npm run lint:fix`_
       **Validate via `npm test`**

**_Is this ready for review?:_** YES
**_Is it a breaking change?:_** NO
